### PR TITLE
fix build warning to [-Wstringop-overread].

### PR DESCRIPTION
**Pull request by cuiziwei1**
## Summary
note/note_driver.c:639:21: warning: 'strlen' reading 1 or more bytes from a region of size 0 [-Wstringop-overread]
  639 |           namelen = strlen(tcb->name);
      |                     ^~~~~~~~~~~~~~~~~

### CODE REVIEW

File: libs/libc/semaphore/sem_init.c [hunk 1]
```diff
@@ -79,7 +79,7 @@ int nxsem_init(FAR sem_t *sem, int pshared, unsigned int value)
   sem->flags = 0;
 
 #ifdef CONFIG_PRIORITY_INHERITANCE
-#  if CONFIG_SEM_PREALLOCHOLDERS > 0
+#  if defined(CONFIG_SEM_PREALLOCHOLDERS) && CONFIG_SEM_PREALLOCHOLDERS > 0
```
Comment (anchao) [inline libs/libc/semaphore/sem_init.c:82]: move below define to `include/semaphore.h`
https://github.com/apache/nuttx/blob/master/sched/semaphore/sem_holder.c#L45-L47

**Review by pkarashchenko — APPROVED** [review]

**Comment by pkarashchenko** [comment]
Please squash the chnages

File: drivers/note/note_driver.c [hunk 2]
```diff
@@ -652,7 +652,10 @@ void sched_note_start(FAR struct tcb_s *tcb)
            */
 
 #if CONFIG_TASK_NAME_SIZE > 0
-          namelen = strlen(tcb->name);
+          if (tcb->name[0] != '\0')
+            {
+              namelen = strlen(tcb->name);
```
[the remark this answers is not in the document]
Comment (cuiziwei1) [reply drivers/note/note_driver.c]: when use **strnlen**，I fount this problem:
note/note_driver.c:674:21: error: 'strnlen' specified bound 32 exceeds source size 0 [-Werror=stringop-overread] 
In fact, it has been determined that CONFIG_TASK_NAME_SIZE>0, but without judging whether tcb->name is empty, the compiler still reports an error, perhaps a false positive from gcc13.

**Review by anchao — CHANGES_REQUESTED** [review]
Please merge `CONFIG_SEM_PREALLOCHOLDERS` patch set into one

**Pull request closed without merging**